### PR TITLE
Make the cursor no longer blinking when move, as same as the effect of iOS platform.

### LINKED ISSUE #105454: Long press blank and drag it to move cursor. The cursor shouldn't blink.
## Steps to Reproduce

The blinking animation of Flutter app looks different with iOS native app when I move the cursor.

There're screen records of Flutter app and iOS native app.

### Flutter

https://user-images.githubusercontent.com/7621572/172211845-a89a9dda-f60a-4c9a-aec3-9e96c1a53409.mov

### iOS Native
https://user-images.githubusercontent.com/7621572/172211895-da8897f1-66b3-4109-b296-e0b709f07a65.mov

### Steps

1. Long press and drag the blank bar.
2. Look at the cursor. The cursor will keep blinking while dragging. But iOS won't.

**flutter doctor**
<details>
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 3.0.1, on macOS 12.4 21F79 darwin-x64, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.67.2)
[✓] Connected device (1 available)
[✓] HTTP Host Availability

• No issues found!
```
</details>

### PR DESCRIPTION
**Pull request by talisk**
Fix #105454 

**Before**

https://user-images.githubusercontent.com/7621572/172214290-893f0868-0138-4c30-b8d0-ed02fada9331.mov

**After**

https://user-images.githubusercontent.com/7621572/172214298-f6a53820-1d65-4162-a21f-eae295c67342.mov

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by talisk** [comment]
@goderbauer @justinmc @Piinks

**Review by LongCatIsLooong — COMMENTED** [review]
Thanks for contributing! Could you sync this pull request to the head of the master branch and see if there are any merge conflicts?

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 1]
```diff
@@ -788,13 +866,6 @@ void main() {
     await tester.pump();
 
     expect(editable, paints
-      ..rrect(
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:791]: Ah I see from the other pull request the gray cursor is removed. In a UIKit app I believe it still exists (you can verify by dragging the blue cursor to a location that's far from the text), it's there to indicate the `TextPosition` the caret will snap to once you let it go.

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 2]
```diff
@@ -2106,17 +2109,19 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
         _lastBoundedOffset = _startCaretRect!.center - _floatingCursorOffset;
         _lastTextPosition = currentTextPosition;
-        renderEditable.setFloatingCursor(point.state, _lastBoundedOffset!, _lastTextPosition!);
+        renderEditable.setFloatingCursor(point.state, _lastBoundedOffset!, _lastTextPosition!, resetLerpValue: _floatingCursorResetController!.value);
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/editable_text.dart:2112]: Is this change for removing the gray cursor? If so see the comment: https://github.com/flutter/flutter/pull/107221#discussion_r917180300

[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/lib/src/widgets/editable_text.dart:2112]: Yep. It's change for removing the gray cursor.

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 3]
```diff
@@ -656,6 +656,84 @@ void main() {
...
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final EditableTextState editableText = tester.state(find.byType(EditableText));
+
+    // Check that the cursor visibility toggles after each blink interval.
+    Future<void> checkCursorBinking() async {
```
[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart]: Done.

File: packages/flutter/test/widgets/editable_text_cursor_test.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:791]: > Ah I see from the other pull request the gray cursor is removed. In a UIKit app I believe it still exists (you can verify by dragging the blue cursor to a location that's far from the text), it's there to indicate the `TextPosition` the caret will snap to once you let it go.

I don't think the UIKit app need the gray cursor. The screen recording in #105454 shows the different between iOS native app and Flutter app. The gray cursor has no longer existed when drag the cursor.

[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:836]: image

> In fact, this PR will fix two differences.
> 
> First, the blue(Color 0xbf2196f3) cursor no longer blinks while dragging. Second, when dragging, the gray(Color 0xff999999) cursor that moves between text is also removed. These are the differences between the Flutter and the iOS native.
> These two ..rrects validate the previous grey(Color 0xff999999) cursor at the bottom.

https://github.com/flutter/flutter/pull/105455#discussion_r892066388

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:836]: You have to drag the floating cursor away from the text to see the gray cursor:

image

It doesn't appear when the blue cursor is in the vicinity but it does appear. It's a useful indication sometimes I guess.

[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:836]: It looks like it is. Can we only keep no blinking changes and make the gray cursor back in this PR? About the behavior of the gray cursor, I might look into it again and maybe fix it in another PR.

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:836]: Of course! Thanks for looking into this.

[the remark this answers is not in the document]
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart:836]: Done

File: packages/flutter/test/widgets/editable_text_cursor_test.dart [hunk 4]
```diff
@@ -834,7 +901,7 @@ void main() {
 
     expect(editable, paints
       ..rrect(
-        rrect: RRect.fromRectAndRadius(
```
Comment (LongCatIsLooong) [inline packages/flutter/test/widgets/editable_text_cursor_test.dart:837]: nit: could you revert the indentation change?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM module intentation

File: packages/flutter/test/widgets/editable_text_cursor_test.dart (hunk 4 above)
Comment (talisk) [reply packages/flutter/test/widgets/editable_text_cursor_test.dart]: Done.

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 5]
```diff
@@ -2184,6 +2184,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
           _floatingCursorResetController!.stop();
           _onFloatingCursorResetTick();
         }
+        // Stop cursor blinking and making it visable.
```
Comment (cbracken) [inline packages/flutter/lib/src/widgets/editable_text.dart:2187]: nit: visible

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 6]
```diff
@@ -2204,6 +2207,8 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         renderEditable.setFloatingCursor(point.state, _lastBoundedOffset!, _lastTextPosition!);
         break;
       case FloatingCursorDragState.End:
+        // Resume cursor blinking
```
Comment (cbracken) [inline packages/flutter/lib/src/widgets/editable_text.dart:2210]: nit: add a trailing period after 'blinking'.

**Review by cbracken — APPROVED** [review]
lgtm modulo a couple nitpicks.

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 6 above)
Comment (talisk) [reply packages/flutter/lib/src/widgets/editable_text.dart]: Done.

**Review by oyefalak — COMMENTED** [review]
good

**Pull request merged**